0004-ledger-plugin-interface/0004-ledger-plugin-interface.md

```diff
@@ -199,82 +204,83 @@ Submit a fulfillment to a ledger. Plugin must be connected, otherwise the promis
 
-#### Event: `receive`
-<code>ledgerPlugin.on('receive', ( **transfer**:[IncomingTransfer](#incomingtransfer) ) ⇒ )</code>
+### Event: `*_transfer`
+<code style="">ledgerPlugin.on('incoming_transfer',
+  (
+    **transfer**:[Transfer](#class-transfer),
+  ) ⇒
+)</code>
 
-Emitted when a transfer is received.
+Emitted on an outgoing/incoming transfer which does not have a condition. 
 
-Note that transfers may be conditional, in which case the `receive` event **DOES NOT** indicate that money has been transferred. If users of the plugin wish to check whether they have gotten money, they MUST check whether the transfer has a condition. If transfers have conditions, the final status will only be known when the [fulfill_execution_condition](#event-fulfill_execution_condition-) or [fulfill_cancellation_condition](#event-fulfill_cancellation_condition-) are emitted.
+This indicates that the funds have already been
+transfered. In order to prevent unexpected incoming funds, a ledger MAY allow users to forbid incoming transfers without
+conditions.
 
-The ledger plugin MUST authenticate the source for all incoming transfers, whether they include money or not.
+If the event is `outgoing_transfer`, then it means you sent the transfer. `incoming_transfer` means somebody sent funds
+to you.
 
-###### Example `transfer`
+### Event: `*_prepare`
+<code style="">ledgerPlugin.on('incoming_prepare',
+  (
+    **transfer**:[Transfer](#class-transfer),
+  ) ⇒
+)</code>
 
-```js
-{
-  id: 'https//ledger.example/transfers/123',
-  account: 'https://ledger.example/accounts/connector',
-  amount: '10',
-  data: Buffer(...),
-  noteToSelf: {},
+Emitted when an outgoing/incoming transfer is prepared.
 
-  // for UTP/ATP support
-  executionCondition: 'cc:0:3:47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU:0',
+Note that the `*_prepare` event **DOES NOT** indicate that money has been transferred. The final status will only be known when either the [*_fulfill](#event-*_fulfill-) or [*_cancel](#event-*_cancel-) events are emitted.
 
-  // for UTP support
-  expiresAt: '2016-05-18T12:00:00.000Z',
-
-  // for ATP support
-  cancellationCondition: 'cc:0:3:47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU:0'
-}
-```
+The ledger plugin MUST authenticate the source for all incoming transfers, whether they include money or not.
 
-For a detailed description of these properties, please see [`IncomingTransfer`](#incomingtransfer).
+If the event is `outgoing_prepare`, then it means you prepared the transfer. `incoming_prepare` means someone prepared
+a transfer to you.
 
-#### Event: `fulfill_execution_condition`
-<code style="">ledgerPlugin.on('fulfill_execution_condition',
+### Event: `*_fulfill`
+<code style="">ledgerPlugin.on('outgoing_fulfill',
   (
     **transfer**:[Transfer](#class-transfer),
     **fulfillment**:Buffer
   ) ⇒
 )</code>
 
-Emitted when a transfer's execution condition has been fulfilled.
+Emitted when an outgoing/incoming transfer is fulfilled.
```

Comment:
Should mention that this is only applicable to transfers with a condition. Kind of implied, but I could certainly see somebody assume that this includes transfers without a condition.